server/server.js

```diff
@@ -0,0 +1,99 @@
+const express = require('express');
+const app = express();
+const cors = require("cors")
+
+app.use(express.json())
+app.use(cors())
+
+app.listen(3000, function () {
+    console.log("Server is running");
+});
+
+const something = {
+	name: "My Name",
+	city: "My City",
+    message: "Hi"
+}
+
+let todos = [
+    { 
+        title: 'Do laundry', 
+        done: false, 
+        id: 1
+    },
+    { 
+        title: 'Make dinner', 
+        done: true, 
+        id: 2
+    },
+];
+
+app.get("/", function (req, res) {
+    res.send(something);
+});
+
+app.get("/todos", function (req, res) {
+    res.send(todos);
+});
+
+let currentId = todos.length + 1;
+
+app.post('/todos', function (req, res) {
+    const receivedTodos = req.body;
+
+    if (!receivedTodos.id) {
+        receivedTodos.id = currentId;
+        currentId++;
+    }
+
+    todos.push(receivedTodos);
+    console.log(req.body);
+	res.sendStatus(201);
+});
+
+function updateId (arr) {
+    arr.forEach((item, index) => {
+        item.id = index + 1;
+        index++;
+    });
+    currentId = arr.length + 1;
+}
+
+app.put('/todos', function (req, res) {
+    todos = todos.filter(item => !item.done);
```

Comment:
signature of put method is `PUT /task/:id` 
so you supposed to have that ID parameter in the path:
```js
app.put('/todos/:id', function (req, res) {
 // how do you get {id} value from `req` ?
}
```
try express documentation here: https://expressjs.com/en/guide/routing.html 
you need learn how to use route parameters. these are also called "path parameters".